edweiss-app/app/(app)/deck/index.tsx

```diff
@@ -0,0 +1,200 @@
+        setSelectionMode(false); // Exit selection mode
+    };
+
+    const cancelSelection = () => {
+        setSelectedDecks([]); // Clear selection
+        setSelectionMode(false); // Exit selection mode
+    };
+
+    const enterSelectionMode = () => {
+        setSelectionMode(true); // Activate selection mode on long press
+    };
+
+    return (
+        <>
+            <RouteHeader title={"Decks"} />
+
+            <TScrollView>
+
+                <FancyTextInput
+                    value={deckName}
+                    onChangeText={n => {
+                        setDeckName(n);
+                        setExistedDeckName(false);
+                    }}
+                    placeholder='My amazing deck'
+                    icon='dice'
+                    label='Deck name'
+                    error={existedDeckName ? 'This name has already been used' : undefined}
+                    multiline
+                    numberOfLines={3}
+                />
+
+                <FancyButton backgroundColor='blue' mt={'md'} mb={'sm'} onPress={call} icon='logo-firebase'>
+                    {t("memento:create-deck")}
+                </FancyButton>
+
+                {selectedDecks.length > 0 && (
+                    <TView mt={'md'} style={{ flexDirection: 'row', justifyContent: 'space-between' }}>
+                        <FancyButton
+                            backgroundColor='red'
+                            onPress={deleteSelectedDecks}
+                            mb={'sm'}
+                            style={{ flex: 1 }}
+                        >
+                            Delete Selected Deck
+                        </FancyButton>
+
+                        <FancyButton
+                            backgroundColor='blue'
+                            onPress={cancelSelection}
+                            style={{ flex: 1 }}
+                        >
+                            Cancel
+                        </FancyButton>
+                    </TView>
+                )}
+
+                <For each={decks}>
+                    {deck =>
+                        <DeckDisplay
+                            key={deck.id}
+                            deck={deck.data}
+                            id={deck.id}
+                            isSelected={selectedDecks.some(selected => selected.name === deck.data.name)}
+                            toggleSelection={toggleDeckSelection}
+                            onLongPress={enterSelectionMode}
+                            selectionMode={selectionMode}
+                        />}
+                </For>
+            </TScrollView>
+        </>
+    );
+};
+
+export default DeckScreen;
+
+export const DeckDisplay: ReactComponent<{ deck: Memento.Deck, id: string; isSelected: boolean; toggleSelection: (deck: Memento.Deck) => void; onLongPress: () => void; selectionMode: boolean; }> = ({ deck, id, isSelected, toggleSelection, onLongPress, selectionMode }) => {
```

Comment:
In my opinion, you could create a new file to put the DeckDisplay in component/memento, in order to modularize the code a bit more.